### ISSUE TITLE
Only first channel added from Prime HDHR xml

### ISSUE THREAD

**Issue by sengsational** [sengsational/cwhelper#49]
Although the log indicates many channels added, only one channel is actually added.

    running scan for tuner 1310D28B-2
    Sat Dec 07 18:03:47 EST 2024 Call Stack: org.cwepg.hr.LineUpHdhr scan -- org.cwepg.hr.TunerHdhr scanRefreshLineUp -- org.cwepg.hr.TunerManager scanRefreshLineUpTm --
    Sat Dec 07 18:03:47 EST 2024 Executing request http://172.16.0.123/lineup.xml?tuning
    finished executing request with 135482 characters received.
    Sat Dec 07 18:03:47 EST 2024 Clearing channels for tuner: 1310D28B-2
    Sat Dec 07 18:03:47 EST 2024 484 channels added to tuner: 1310D28B-2
    Sat Dec 07 18:03:47 EST 2024 1310D28B-2 - useExistingFile:true (#1) vChannelTuner:true 

One channel added:

    channelName="0.0" alphaDescription="LOCSTJ" input="1" protocol="auto" favorite="false" sourceId="" tuner="1310D28B-2" deviceId="319869579" tunerType="2" channelDescription="3 LOCSTJ" channelVirtual="3" frequency="0" priority="5792000.0" alternates="[]"

The count in the log comes from a basic counter https://github.com/sengsational/cwhelper/blob/45c04fb829b461ae0000fb27f95086c9ead48b84/CwHelperC/src/org/cwepg/hr/LineUpHdhr.java#L183

The channel add is not validated, so bad channels are skipped. https://github.com/sengsational/cwhelper/blob/45c04fb829b461ae0000fb27f95086c9ead48b84/CwHelperC/src/org/cwepg/hr/LineUpHdhr.java#L180

It's unknown why some XML files process completely, and others process only the first channel.

**Comment by sengsational** [comment]
The file that includes Modulation "auto" does not create a legit `ChannelDigital` object.

    Sun Dec 08 14:43:02 EST 2024 Successfully added channel    [<GuideNumber>68.6</GuideNumber><GuideName>TOONS</GuideName><TransportStreamID>4055</TransportStreamID><Modulation>8vsb</Modulation><Frequency>521000000</Frequency><ProgramNumber>9</ProgramNumber>   <VideoCodec>MPEG2</VideoCodec><AudioCodec>AC3</AudioCodec><URL>http://192.168.1.16:5004/auto/v68.6</URL>]
    Sun Dec 08 14:43:55 EST 2024 Not able to add channel using [<GuideNumber>1995</GuideNumber><GuideName>CSCT</GuideName>                                            <Modulation>auto</Modulation><Frequency>225000000</Frequency><ProgramNumber>1503</ProgramNumber><VideoCodec>MPEG2</VideoCodec><AudioCodec>AC3</AudioCodec><URL>http://172.16.0.45:5004/auto/v1995</URL>]

**Comment by sengsational** [comment]
I made a change so that when GuideNumber does not contain a ".", the virtualChannel gets populated with the contents of GuidNumber, and virtualSubchannel gets populated with "0".

https://github.com/sengsational/cwhelper/commit/c3427c6318b174e86a4c2e7d9d3a1d015d98896e

Uploading shortly as 1044.

Note that if you have the ability to test in Eclipse, I added a main method to LineUpHdhr.java that lets you define a local XML file name to see if it makes the expected channels.
